crates/rattler_lock/src/parse/mod.rs

```diff
@@ -42,38 +45,37 @@ pub enum ParseCondaLockError {
     LocationToUrlConversionError(#[from] file_url::FileURLParseError),
 }
 
-impl FromStr for LockFile {
-    type Err = ParseCondaLockError;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        // First parse the document to a `serde_yaml::Value`.
-        let document: Value = serde_yaml::from_str(s).map_err(ParseCondaLockError::ParseError)?;
-
-        // Read the version number from the document
-        let version: FileFormatVersion = document
-            .get("version")
-            .ok_or_else(|| {
+pub(crate) fn from_str_with_base_directory(
```

Comment:
This would as be a API break :) cc @baszalmstra

Reply:
Yeah one is a user-facing API break the other is a programmatic one, so we should bump the version in the Cargo.toml as well. :)